### PR TITLE
[MO] - [system test] -> Pod Security policies 

### PR DESCRIPTION
**Pull request by see-quick**
### Type of change

- Enhancement / new feature

### Description

This PR adds a test cases for the `Pluggable Pod Security Providers` feature (i.e., https://github.com/strimzi/strimzi-kafka-operator/pull/7089). 

PodSecurityPoliciesST provides tests for Pod Security policies. In short, Pod security profiles are a mechanism used in Pods or containers, which may prohibit some operations (e.g., running only as a non-root user, allowing only some Volume types etc.).

Reason why is this test suite has to run in complete isolation (i.e., {@link IsolatedSuite}) is that we need to modify Cluster Operator configuration, specifically env {@code STRIMZI_POD_SECURITY_PROVIDER_CLASS} to restricted.

Test cases are designed to verify the standard behaviour of Pod Security policies. Specifically, (i.) we check if containers such as Kafka, ZooKeeper, EntityOperator, and KafkaBridge have correctly set .securityContext (ii.) then we check if these resources working and are stable with exchanging messages.

### Checklist

- [x] Write tests
- [x] Make sure all tests pass

### CODE REVIEW

**Review by im-konge — APPROVED** [review]

**Review by scholzj — COMMENTED** [review]
Pod Security Policies is the name of the Kubernetes feature which is being removed (https://kubernetes.io/docs/concepts/security/pod-security-policy/). The new feautre is called Pod Security Admission, eventually Pod Security Standards / Pod Security Profiles refers to the actual profiles the pods might match. Or Pod Security Providers is how we called the Strimzi plugins. We should name it accordingly to avoid any confusion wth the deprecated Kube APIs.

File: systemtest/src/main/java/io/strimzi/systemtest/annotations/RequiredMinKubeOrOpenshiftVersionCondition.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        if ((isOcp && Double.parseDouble(cluster.client().clusterKubernetesVersion()) >= ocpBasedKubeVersion) ||
+            Double.parseDouble(cluster.client().clusterKubernetesVersion()) >= kubeVersion) {
+            return ConditionEvaluationResult.enabled("Test is enabled");
+        } else {
+            LOGGER.info("{} is @RequiredMinKubeOrOpenshiftVersion with type of cluster: {} and version {}, but the running on cluster with {}: Ignoring {}",
+                extensionContext.getDisplayName(),
+                cluster.cluster().getClass().toString(),
+                isOcp ? ocpBasedKubeVersion : kubeVersion,
+                cluster.client().clusterKubernetesVersion(),
+                extensionContext.getDisplayName()
```
Comment (Frawless) [inline systemtest/src/main/java/io/strimzi/systemtest/annotations/RequiredMinKubeOrOpenshiftVersionCondition.java:39]: Guess we don't need it twice in one log message

**Review by Frawless — APPROVED** [review]
Just one nit + Jakub's comment

**Comment by see-quick** [comment]
> Pod Security Policies is the name of the Kubernetes feature which is being removed (https://kubernetes.io/docs/concepts/security/pod-security-policy/). The new feautre is called Pod Security Admission, eventually Pod Security Standards / Pod Security Profiles refers to the actual profiles the pods might match. Or Pod Security Providers is how we called the Strimzi plugins. We should name it accordingly to avoid any confusion wth the deprecated Kube APIs.

I have changed it to **Pod Security profiles**.

**Review by scholzj — COMMENTED** [review]
I do not see you to configure the enforcement of the restricted profile on the namespace level. So does this purely test the security context from the pods? Should we also enforce the profile? (https://kubernetes.io/docs/concepts/security/pod-security-admission/#pod-security-admission-labels-for-namespaces)

File: test/src/main/java/io/strimzi/test/k8s/KubeClient.java [hunk 2]
```diff
@@ -91,6 +91,10 @@ public void createNamespace(String name) {
         client.namespaces().resource(ns).createOrReplace();
     }
 
+    public void createNamespace(final Namespace namespace) {
```
Comment (scholzj) [inline test/src/main/java/io/strimzi/test/k8s/KubeClient.java:94]: Should it be called `createOrReplaceNamespace` then?

**Review by scholzj — COMMENTED** [review]
One nit. LGTM otherwise.

File: test/src/main/java/io/strimzi/test/k8s/KubeClient.java (hunk 2 above)
Comment (see-quick) [reply test/src/main/java/io/strimzi/test/k8s/KubeClient.java:94]: > Should it be called `createOrReplaceNamespace` then?

Yes, will do.

**Review by scholzj — APPROVED** [review]
LGTM assuming the tests pass.

**Pull request merged**